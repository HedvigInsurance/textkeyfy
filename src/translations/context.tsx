import * as React from 'react'
import { Language } from './types'

export const TranslationsContext = React.createContext<Language>({
  textKeys: {},
})
