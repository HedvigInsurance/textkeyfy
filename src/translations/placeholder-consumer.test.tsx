import toJson from 'enzyme-to-json'
import * as React from 'react'

import { TranslationsContext } from './context'
import {
  replacePlaceholders,
  TranslationsPlaceholderConsumer,
} from './placeholder-consumer'

import { mount } from 'enzyme'

test('should replace placeholders correctly', () => {
  const TestElement = <div />

  const elementAndText = replacePlaceholders(
    {
      testElement: TestElement,
    },
    '{testElement} mock',
  )
  const elementAndTextWrapper = mount(<div>{elementAndText}</div>)

  expect(elementAndTextWrapper.containsMatchingElement(TestElement)).toBe(true)
  expect(toJson(elementAndTextWrapper)).toMatchSnapshot()

  const elementAndTextWithoutSpace = replacePlaceholders(
    {
      testElement: TestElement,
    },
    '{testElement}mock',
  )
  const elementAndTextWithoutSpaceWrapper = mount(
    <div>{elementAndTextWithoutSpace}</div>,
  )

  expect(
    elementAndTextWithoutSpaceWrapper.containsMatchingElement(TestElement),
  ).toBe(true)
  expect(toJson(elementAndTextWithoutSpaceWrapper)).toMatchSnapshot()

  const withUnmatchedIndex = replacePlaceholders(
    {
      testing: 'testing',
    },
    '{somethingelse} mock',
  )
  const withUnmatchedIndexWrapper = mount(<div>{withUnmatchedIndex}</div>)

  expect(withUnmatchedIndexWrapper.text().includes('{somethingelse}')).toBe(
    true,
  )
  expect(toJson(withUnmatchedIndexWrapper)).toMatchSnapshot()

  const TestElement2 = <div>mock</div>

  const withMultiplePlaceholders = replacePlaceholders(
    {
      testElement: TestElement,
      testElement2: TestElement2,
      testing: 'testing',
    },
    '{testElement} mock {testing} mock {testElement2}',
  )
  const withMultiplePlaceholdersWrapper = mount(
    <div>{withMultiplePlaceholders}</div>,
  )

  expect(
    withMultiplePlaceholdersWrapper.containsMatchingElement(TestElement),
  ).toBe(true)
  expect(
    withMultiplePlaceholdersWrapper.containsMatchingElement(TestElement2),
  ).toBe(true)
  expect(toJson(withMultiplePlaceholdersWrapper)).toMatchSnapshot()

  const TestElement3 = <div>mock</div>

  const withUnderscores = replacePlaceholders(
    {
      mock_key: TestElement3,
    },
    '{mock_key} foo bar baz',
  )

  const withUnderscoresWrapper = mount(<div>{withUnderscores}</div>)

  expect(withUnderscoresWrapper.containsMatchingElement(TestElement3)).toBe(
    true,
  )
  expect(toJson(withUnderscoresWrapper)).toMatchSnapshot()
})

test('it should render correctly', () => {
  const TestElement = <div />
  const wrapper = mount(
    <TranslationsContext.Provider
      value={{ textKeys: { mock: '{testElement} mock' } }}
    >
      <TranslationsPlaceholderConsumer
        textKey="mock"
        replacements={{
          testElement: TestElement,
        }}
      >
        {(nodes) => <div>{nodes}</div>}
      </TranslationsPlaceholderConsumer>
    </TranslationsContext.Provider>,
  )

  expect(wrapper.containsMatchingElement(TestElement)).toBe(true)

  expect(toJson(wrapper)).toMatchSnapshot()
})
