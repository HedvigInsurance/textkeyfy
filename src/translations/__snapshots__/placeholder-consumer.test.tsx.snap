// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
<Component
  replacements={
    Object {
      "testElement": <div />,
    }
  }
  textKey="mock"
>
  <Component
    textKey="mock"
  >
    <div>
      <Component
        key="0"
      >
        <div />
      </Component>
       mock
    </div>
  </Component>
</Component>
`;

exports[`should replace placeholders correctly 1`] = `
<div>
  <Component
    key="0"
  >
    <div />
  </Component>
   mock
</div>
`;

exports[`should replace placeholders correctly 2`] = `
<div>
  <Component
    key="0"
  >
    <div />
  </Component>
  mock
</div>
`;

exports[`should replace placeholders correctly 3`] = `
<div>
  {somethingelse}
   mock
</div>
`;

exports[`should replace placeholders correctly 4`] = `
<div>
  <Component
    key="0"
  >
    <div />
  </Component>
   mock 
  <Component
    key="2"
  >
    testing
  </Component>
   mock 
  <Component
    key="4"
  >
    <div>
      mock
    </div>
  </Component>
</div>
`;

exports[`should replace placeholders correctly 5`] = `
<div>
  <Component
    key="0"
  >
    <div>
      mock
    </div>
  </Component>
   foo bar baz
</div>
`;
